rewrite-java/src/main/java/org/openrewrite/java/search/FindMethods.java

```diff
@@ -25,6 +25,24 @@
 import static java.util.Collections.emptyList;
 import static java.util.Collections.singletonList;
 
+/**
+ * A Java search visitor that will return a list of matching method invocations within the abstract syntax tree.
+ * This visitor is configured with a single parameter that represents a method signature using the AspectJ pointcut
+ * syntax. The signature syntax allows for wildcard matches against the receiver, the method name, and the method
+ * arguments.
+ * <P><P>
+ * <PRE>
+ *     EXAMPLES:
+ *
+ *      * *(..)                                 - All method invocations
+ *      java.util.* *(..)                       - All method invocations to classes belonging to java.util (including sub-packages)
+ *      java.util.Collections *(..)             - All method invocations on java.util.Collections class
+ *      java.util.Collections unmodifiable*(..) - All method invocations starting with "unmodifiable" on java.util.Collections
+ *      java.util.Collections min(..)           - All method invocations for all overloads of "min"
+ *      java.util.Collections emptyList()       - All method invocations on java.util.Collections.emptyList()
+ *      my.org.MyClass *(boolean, ..)           - All method invocations where the first arg is a boolean in my.org.MyClass
```

Comment:
Several people who have seen this syntax for the first time mistakenly assumed that the first part was the _return type_ of the method, rather than the type of the method's receiver. 

To try and combat this particular misunderstanding in [our docs](https://docs.openrewrite.org/java/refactoring-java-source-code#matching-method-signatures-with-methodmatcher) I described the functionality of the syntax in the same order as its components, so receiver first, then method name, then arguments.
Is this the ideal way of doing it? I'm not sure, but I suspect it's an improvement.

Reply:
yeah, I like how you explicitly call out the sections of the expression. I will align my Javadocs to do that as well.